specs/slider/slider.md

```diff
@@ -0,0 +1,181 @@
+- Implement your own background and slider track UI, default provided
+- Handle the `slider`'s `onValueChange` callback to be notified when the value has changed.
+- Customize the tick marks along the track
+- Include slider in form based operations
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/slider)
+- [Material UI](https://material-ui.com/components/slider/)
+
+---
+
+### API
+
+*Component Name*:
+- `fast-slider`
+
+*Props/Attrs*
+- `min` - minimum allowed value for the slider
+- `max` - maximum allowed value for the slider
+- `step` - limits the values of the `slider` to increments of the `step` value added to the minimum value of the 
+`slider`'s total range.  The default value is 1. The minimum and maximum values of a `slider`'s range are always valid results regardless of the `step` prop. The `step` prop is used as the value for incrementing the thumb by pressing the arrow keys.
+ - `value` - Allows authors to specify the initial selected range of the `slider`.  It defaults to a (step constrained) value at the midpoint on the `slider`'s total range. 
+
+*Events*
+- `change` - raise the change event for external parties to be informed of the value change.
+
+*Slots*
+- `background-track` - background area encompassing the track and labels
+- `slider-track` - the horizontal or vertical track along which the thumb slides
+- `thumb` - the control the user drags along the track to change the value of the slider
+
+### Anatomy and Appearance
+**Structure:**
+
+```html
+  <div
+    role="slider"
+  >
+    <input type="range" 
+      id="${x => x.id}"
+      min="${x => x.min}"
+      max="${x => x.max}"
+      step="${x => x.step}"
+      value="${x => x.value}"
+    >
+    <slot name="label">
+      <div>${x => x.label}</div>
+    </slot>
+    <slot name="background-track">
+      <div></div>
+    </slot>
+    <slot name="slider-track">
+      <div></div>
+    </slot>
+    <slot name="thumb">
+      <div></div>
+    </slot>
+    <slot></slot>
+  </div>
+```
+
+## Implementation
+
+```html
+<fast-slider
+    id="slider1"
+    value="32"
+    min="0"
+    max="44"
+    step="1"
+>
+    <h2 slot="label"><b>Density</b></h2>
+    <div slot="thumb"><img src="..."/></div>
+    <fast-slider-label
+      label="32"
+      show-mark="true"
+    >
+    </fast-slider-label>
+</fast-slider>
+```
+
+---
+
+### Accessibility
+
+*Slider* implements the recommended keyboard navigation scheme described [W3C aria-practices](http://w3c.github.io/aria-practices/examples/slider/multithumb-slider.html).
+
+- aria-valuenow will be set as the slider value changes
+- right and up arrow move the slider in increasing value and left and down arrows move it in decreasing increments
+- *Form Input*
+
+
+### Globalization
+Slider should render a mirror view in rtl for example:
+![](./images/slider-rtl.png) 
+
+- Strings passed into the component should already be localized

```

Comment:
Could you detail the expectation for how the upper/lower thumbs respond in terms of positioning and keyboarding?

Reply:
changed this to just refer to thumb. We can maybe talk offline but thinking this component should go in as simple as possible and can be expanded upon in subsequent versions.

Reply:
Did this get deleted all up? If we're planning on including upper and lower values via thumbs I'd like that represented in the spec. We can still implement that as "next steps" and not include in the initial version, but I think it's important to capture and understand the full functionality so we don't paint ourselves into a corner with an API.

Reply:
Examples of this can be seen in something like tree view where we detail that we are *not* initially implementing certain functionality.

Reply:
Yes, I will be adding dual thumb back in per our conversation

Reply:
I added back in modes but changed the names slightly. I also added an * to state we should only implement single-value to start.